src/scripts/blocks/countdown-timer/index.js

```diff
@@ -0,0 +1,84 @@
+import classnames from 'classnames';
+import DisplayComponent from './DisplayComponent';
+
+const { __ } = wp.i18n;
+const { registerBlockType } = wp.blocks;
+const { Component, Fragment } = wp.element;
+const {
+  InspectorControls, RichText,
+} = wp.editor;
+const {
+  PanelBody, Button, DateTimePicker, ColorPicker,
+} = wp.components;
+
+const { PostMediaSelector } = benenson.components;
+
+registerBlockType('benenson/block-countdown', {
+  title: __('Countdown timer', 'benenson'),
+  description: __('Set a future date to countdown to.', 'benenson'),
+  icon: 'download',
+  category: 'benenson',
+  supports: {
+  },
+
+  attributes: {
+    title: {
+      type: 'string',
+    },
+    date: {
+      type: 'string',
+    },
+    backgroundColor: {
+      type: 'string',
+    },
+    backgroundId: {
+      type: 'interger',
+    },
+    backgroundUrl: {
+      type: 'string',
+    },
+    textColor: {
+      type: 'string',
+    },
+  },
+  edit: DisplayComponent,
+
+  save({ attributes }) {
+    const {
+      date = new Date(),
+      title = '',
+      backgroundColor = '',
+      backgroundUrl = '',
+      textColor = '',
+    } = attributes;
+
+    const styles = {
+      backgroundImage: `url('${backgroundUrl}')`,
+      backgroundColor,
+    };
+
+    const classes = classnames('countdownTimer', {
+      [`is-${textColor}`]: !!textColor,
+    });
+
+    return (
+      <div className={ classes } style={ styles} data-date={ date }>
+        <h2 className="countdownTimer-title">{ title }</h2>
+        <div className="countdownTimer-items">
+          <div className="countdownTimer-item countdownTimer-days">
```

Comment:
Because the values for each are initially set to `00` and hidden, there's a noticeable FOUC while the scripts are loading. If anything goes wrong with the script loading, then users will see something like this, which isn't ideal:
![](http://bigbite.im/u0Ltvf+)